### ISSUE TITLE
Missing options in pytest configuration file (pytest.ini)

### ISSUE THREAD

**Issue by rileyjmurray** [pyGSTio/pyGSTi#361]
Many of our tests are in classes titled ``<my_class_name>Tester``. By default, PyTest expects a naming convention of the form ``Test<my_class_name>``. PyTest can be configured to automatically discover the tests we have by an appropriate change of ``pytest.ini``. Making this change by branching off from develop at [this commit](https://github.com/pyGSTio/pyGSTi/tree/7ff0aae372ec8bc82f4832fbe526988b08346203) _seems to_ result in more errors when running ``pytest test/unit``.

We should definitely make this change to ``pytest.ini``. I'm filing this issue so that we can track the consequences of making this change, including any specific failing tests that need to be addressed.

### What I've seen so far
For the curious, here are the errors I get with ``pytest test/unit`` immediately after branching from the aforementioned commit:
<img width="1421" alt="Pasted Graphic 6" src="https://github.com/pyGSTio/pyGSTi/assets/10662932/0b43e8dc-dd4b-420e-8205-bda8204069a7">
I'll note weirdness where PyTest _did_ find some tests with the ``Tester`` suffix. However, here are the errors I get when running ``pytest test/unit`` after changing ``pytest.ini``:
<img width="1595" alt="image" src="https://github.com/pyGSTio/pyGSTi/assets/10662932/c0e1f8bd-a5bb-4780-a751-bea174b81b68">
The screenshot above suggests that making the proposed change to ``pytest.ini`` helps PyTest discover another 25 tests that are failing.

**Comment by rileyjmurray** [comment]
I've done some looking into why PyTest found some classes with the ``Tester`` suffix before I changed ``pytest.ini``.  I think this has something to do with class inheritance structures.

Here are the full names used in definitions of relevant classes.
* ``QiboEvotypeTester(BaseCase)``
    * where ``BaseCase`` inherits from ``unittest.TestCase`` and is defined in ``..util.py``.
* ``InterpygateConstructionTester(BaseCase)``
    *  using the same ``BaseCase`` from above.
* ``LinearOpTester(OpBase)``
    * ``OpBase`` inherits directly from ``object`` and is defined earlier in the same file.
* ``RawObjectiveFunctionTester(object)``.
* ``TimeIndependentMDSObjectiveFunctionTester(ObjectiveFunctionData)``
    *  ``ObjectiveFunctionData`` inherits directly from ``object`` and is defined earlier in the same file.
* ``TimeDependentMDSObjectiveFunctionTester(ObjectiveFunctionData)``
    *  using the same ``ObjectiveFunctionData`` from above.

So it seems that with the current ``pytest.ini``, PyTest automatically finds classes with the ``Tester`` suffix if (and only if) the class inherits from ``unittest.TestCase``. If we change ``pytest.ini`` then PyTest finds all classes with the ``Tester`` suffix.

(For reproducibility's sake, I'll note that I'm using PyTest 7.4.3.)

**Comment by rileyjmurray** [comment]
I think tests are spuriously passing on GitHub actions because it's using an older version of an optional dependency called "qibo". Here's a snippet from a passing CI run:
https://github.com/pyGSTio/pyGSTi/actions/runs/6678968840/job/18150581137#step:7:116
It's using qibo 0.1.7.

Using qibo 0.2.1 causes tests to fail [at this commit](https://github.com/pyGSTio/pyGSTi/tree/7ff0aae372ec8bc82f4832fbe526988b08346203) without any changes to ``pytest.ini``.

Looks like I would have gotten the older (correct / safe) version of qibo if I installed pyGSTi with ``pip install -e .['testing']``. Unfortunately for me, I installed pyGSTi as follows:
```
pip install -r requirements.txt
pip install -r optional-requirements.txt
pip install -e .
```
The difference is that ``optional_requirements.txt`` doesn't enforce a version bound on ``qibo``.
